src/buffer/out/textBuffer.cpp

```diff
@@ -2810,16 +2810,14 @@ PointTree TextBuffer::GetPatterns(const til::CoordType firstRow, const til::Coor
             // match and the previous match, so we use the size of the prefix
             // along with the size of the match to determine the locations
             til::CoordType prefixSize = 0;
-            for (const auto parsedGlyph : Utf16Parser::Parse(i->prefix().str()))
+            for (const auto str = i->prefix().str(); const auto& glyph : til::utf16_iterator{ str })
             {
-                const std::wstring_view glyph{ parsedGlyph.data(), parsedGlyph.size() };
                 prefixSize += IsGlyphFullWidth(glyph) ? 2 : 1;
             }
             const auto start = lenUpToThis + prefixSize;
             til::CoordType matchSize = 0;
-            for (const auto parsedGlyph : Utf16Parser::Parse(i->str()))
+            for (const auto str = i->str(); const auto& glyph : til::utf16_iterator{ str })

```

Comment:
`str` needs to be assigned to a variable first, because a range for loop like this:
```
for (init-statement; range-declaration : range-expression)
{
    loop-statement
}
```

gets translated to this:
```
{
    #init-statement#
    auto && __range = range-expression ;
    auto __begin = __range.begin();
    auto __end = __range.end();
    for ( ; __begin != __end; ++__begin)
    {
        range-declaration = *__begin;
        loop-statement
    }
}
```

and if the `range-expression` is a function call like this:
```
auto && __range = til::utf16_iterator{ i->str() };
```

then `i->str()` would go out of scope at the end of the line.

Reply:
Very clever.